lib/assert.js

```diff
@@ -931,6 +933,78 @@ assert.doesNotMatch = function doesNotMatch(string, regexp, message) {
   internalMatch(string, regexp, message, doesNotMatch);
 };
 
+const noop = () => {};
+
+const mustCallChecks = [];
+
+let attachExitHandler = true;
+
+function runCallChecks(exitCode) {
+  for (const context of mustCallChecks) {
+    if (context.actual !== context.exact) {
+      context.error = `Mismatched ${context.name} function calls. ` +
+        `Expected exactly ${context.exact}, actual ${context.actual}.`;
+      const console = require('console');
+      console.error(context.error);
+      console.error(context.stackTrace);
```

Comment:
This line should now be obsolete (`console.error(context.error)` uses `util.inspect` on the passed through error):

```suggestion
```